src/game/level_update.c

```diff
@@ -1033,9 +1033,16 @@ s32 play_mode_paused(void) {
         if (gDebugLevelSelect) {
             fade_into_special_warp(WARP_SPECIAL_LEVEL_SELECT, 1);
         } else {
+        #ifdef DEATH_ON_EXIT_COURSE
+            level_trigger_warp(gMarioState, WARP_OP_DEATH);
+            fade_into_special_warp(WARP_SPECIAL_NONE, 0);
```

Comment:
I don't think you want this line. In general, what I would suggest doing for exit course death is copy the unpausing code exactly as-is, and follow it up with the `level_trigger_warp` call. This way it should function pretty much exactly as a standard death.